tests/internal/test_auto.py

```diff
@@ -1,6 +1,9 @@
+import sys
+
 import pytest
 
 
+@pytest.mark.skipif(sys.version_info < (3, 12, 5), reason="Python < 3.12.5 eagerly loads the threading module")
```

Comment:
"sort of", this should be broken on `main` but... isn't... but somehow moving to dependency `wrapt` causes this to always fail.

The reason is because Python 3.12 started eagerly loading `threading`, which means we cannot unload to save our own copy of the module. This is a known issue that is being fixed and released in Python 3.12.5

It does mean this test might fail when 3.12.5 is released.